dbinspector/src/main/kotlin/im/dino/dbinspector/ui/databases/NavigatorIntentFactory.kt

```diff
@@ -0,0 +1,69 @@
+package im.dino.dbinspector.ui.databases
+
+import android.app.Activity
+import android.content.ActivityNotFoundException
+import android.content.Context
+import android.content.Intent
+import android.widget.Toast
+import androidx.core.app.ShareCompat
+import androidx.core.content.FileProvider
+import im.dino.dbinspector.R
+import im.dino.dbinspector.domain.database.models.DatabaseDescriptor
+import im.dino.dbinspector.ui.schema.SchemaActivity
+import im.dino.dbinspector.ui.shared.Constants
+import timber.log.Timber
+import java.io.File
+
+internal class NavigatorIntentFactory(
+    private val context: Context
+) {
+
+    fun showSchema(database: DatabaseDescriptor) =
+        context.startActivity(
+            Intent(context, SchemaActivity::class.java)
+                .apply {
+                    putExtra(Constants.Keys.DATABASE_PATH, database.absolutePath)
+                    putExtra(Constants.Keys.DATABASE_NAME, database.name)
+                }
+        )
+
+    fun showImportChooser() =
+        Intent.createChooser(
+            Intent(Intent.ACTION_OPEN_DOCUMENT).apply {
+                addCategory(Intent.CATEGORY_OPENABLE)
+                flags = Intent.FLAG_GRANT_READ_URI_PERMISSION
+//                    type = "application/vnd.sqlite3"
+//                    type = "application/x-sqlite3"
```

Comment:
@JonatanPlesko These are actually correct mime types for Sqlite. Android has issues with these. I would like to have it prepared and not commented out in case that Android fixes it.
Will be coded in properly.

Reply:
I suspected that was the case. I had a similar issue with `application/zip` matching some zip's but not all. I'm doubtful they'll fix it, but I welcome your optimism.

Reply:
@JonatanPlesko I agree with your pessimism, but I'll prepare the code and forget about it. Everything is better than comments.